### ISSUE TITLE
make `cursor_trail` work on the same command prompt only

### ISSUE THREAD

**Issue by xuhdev** [kovidgoyal/kitty#8147]
**Is your feature request related to a problem? Please describe.**

Currently `cursor_trail` always shows cursor movement once the distance is beyond a certain threshold.

However, in most cases when the cursor makes a big movement, my eyes either know where
it will land or don't care. For example,

- After typing in a long command and pressing "Enter", I know it will show up in the next command line prompt.
- When a program outputs long segments of text, I don't care where the cursor is.

However, the unnecessary animation in these scenarios is distractive.

The moments I see cursor_trail is only needed when I move cursor within a prompt
buffer. For example, I may press `Alt-F` to jump a word, and the word has an
unknown length to my eyes until I look carefully.

I tried to adjust `cursor_trail_start_threshold` and `cursor_trail_decay`, but
they don't seem to be able to address the problem.

**Describe the solution you'd like**
Add a new option `cursor_trail_scenario` to specify when to enable cursor
movement.

**Comment by wwqwq2313** [comment]
Hello, we tried to solve the issue.

This is what we did:

Added a new cursor_trail_scenario option to control when the cursor trail is shown. Updated the cursor trail logic to check this option before rendering.

You can review changes in this commit: https://github.com/wwqwq2313/kovidgoyal-kitty-8147/commit/3f6821ef45bf9104c621f082b41e3d6cdd29c120.


> [!CAUTION]
> Disclaimer: The concept of solution was created by AI and you should never copy paste this code before you check the correctness of generated code. Solution might not be complete, you should use this code as an inspiration only.

---

Latta AI seeks to solve problems in open source projects as part of its mission to support developers around the world. Learn more about our mission at https://latta.ai/ourmission . If you no longer want Latta AI to attempt solving issues on your repository, you can block this account.

**Comment by kovidgoyal** [comment]
@jinhwanlazy please review.

**Comment by jinhwanlazy** [comment]
While I understand what you're trying to achieve, I'm not sure if we can reliably implement this because the terminal doesn't know the semantics of the movement.

Technically we could detect whether a cursor movement was triggered by a control sequence (like a newline character) versus direct cursor positioning commands, the behavior would vary in many applications and I think it would look even more annoying.

**Comment by xuhdev** [comment]
Sorry I didn't write the code. It's a feature request and it's a bot that wrote the changes...

**Comment by kovidgoyal** [comment]
@jinhwanlazy I think the OP is asking that when the main screen is active and the cursor is in an output region the trail be suppressed. This can be detected fairly reliably at least for shells that implement shell integration. However it will break for shells that dont implement it. Basically the logic would be something like, if the last prompt marking command indicates we are in output mode and the main screen is active, suppress the trail otherwise keep it on. And if no prompt marks are received keep it on. I dont use trails so I am not the right person to comment on how well it works, but that is how I would approach this.

**Comment by jinhwanlazy** [comment]
> if the last prompt marking command indicates we are in output mode

Can you tell me where I should look for this?

**Comment by kovidgoyal** [comment]
On Wed, Dec 25, 2024 at 08:15:23PM -0800, Jinhwan Choi wrote:
> > if the last prompt marking command indicates we are in output mode
>
> Can you tell me where I should look for this?

See screen_cursor_at_a_shell_prompt()
You probably want a slightly different function that returns -2 for
UNKNOWN_PROMPT_KIND. You can probably change that function to do that,
check its call sites to make sure its safe, if not write a new function.

**Comment by Joe-Roser** [comment]
I would definitely appreciate only having cursor_trail on tab completion

**Comment by simonwiles** [comment]
Yeah, I would love to have cursor_trail when editing a command at the command line, but I find it very distracting at all other times.

**Comment by EpicurianCoder** [comment]
Hey @kovidgoyal and @jinhwanlazy. Is this still open and getting worked on? 
I use Kitty a lot and have been looking for a project to contribute to and I wanted to perhaps contribute to an issue that seems like a good entry point before getting more involved. I can look at changing the `screen_cursor_at_a_shell_prompt()` function in the way you mentioned above, as well as just get generally familiar with it and possibly find an alternative solution.

**Comment by kovidgoyal** [comment]
no,, no one is working on it.
